lib/PuppeteerSharp.ruleset

```diff
@@ -1,90 +1,90 @@
+    <Rule Id="CA1304" Action="Error" /> <!-- Error CA1304: The behavior of 'string.ToLower()' could vary based on the current user's locale settings. Replace this call in 'JSHandle.ToString()' with a call to 'string.ToLower(CultureInfo)'. (CA1304) -->
+    <Rule Id="CA1305" Action="Error" /> <!-- String.Format with culture-->
+    <Rule Id="CA1720" Action="Error" /> <!-- Error SA1720: Name contains type-->
+    <Rule Id="CA1721" Action="Error" /> <!-- Error CA1721: The property name 'DefaultArgs' is confusing given the existence of method 'GetDefaultArgs'. Rename or remove one of these members. (CA1721)-->
+    <Rule Id="CA1724" Action="None" /> <!-- Error CA1724: The type name Extensions conflicts in whole or in part with the namespace name 'Microsoft.Extensions'. Change either name to eliminate the conflict. (CA1724) -->
+    <Rule Id="CA1806" Action="Error" /> <!-- Error CA1806: DownloadAsync calls Chmod but does not use the HRESULT or error code that the method returns. This could lead to unexpected behavior in error conditions or low-resource situations. Use the result in a conditional statement, assign the result to a variable, or pass it as an argument to another method. (CA1806)-->
+    <Rule Id="CA1816" Action="Error" /> <!-- Error CA1816: Change Connection.Dispose() to call GC.SuppressFinalize(object). This will prevent derived types that introduce a finalizer from needing to re-implement 'IDisposable' to call it. (CA1816) -->
+    <Rule Id="CA2000" Action="Error" /> <!-- Error CA2000: Call System.IDisposable.Dispose on object created by 'new Process()' before all references to it are out of scope. (CA2000) -->
+    <Rule Id="CA2008" Action="None" /> <!-- Error ca2008: Pass TaskScheduler-->
+    <Rule Id="CA2200" Action="Error" /> <!-- Error CA2200: Re-throwing caught exception changes stack information. (CA2200)  -->
+    <Rule Id="CA2213" Action="Error" /> <!-- Error CA2213: 'WebSocketTransport' contains field '_readerCancellationSource' that is of IDisposable type 'CancellationTokenSource', but it is never disposed. Change the Dispose method on 'WebSocketTransport' to call Close or Dispose on this field. (CA2213) -->
+    <Rule Id="CA2220" Action="None" /> <!-- Error CA2229: Add a constructor to SelectorException with the following signature: 'protected SelectorException(SerializationInfo info, StreamingContext context)'. (CA2229) (PuppeteerSharp)-->
+    <Rule Id="CA2229" Action="Error" /> <!-- Error CA2229: Add a constructor to SelectorException with the following signature: 'protected SelectorException(SerializationInfo info, StreamingContext context)'. (CA2229)-->
+    <Rule Id="CA2237" Action="None" /> <!-- Error CA2237: Add [Serializable] to TargetClosedException as this type implements ISerializable (CA2237)-->
+  </Rules>
+  <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
+    <Rule Id="CA1303" Action="None" />
+  </Rules>
```

Comment:
Can we avoid changing the whole file?

Reply:
I will take care of this next time, since I was making changes in Ruleset UI(VS2019) comments are vanished don't know why I'm rewriting them, but I figured out if I set rules using Notepad++ I don't face any such problems

Reply:
This is still wrong. Maybe a line-ending change?

Reply:
Sorry @kblok  I didn't get this do we need to add a line ending (/n) new line character ?

Reply:
Could you try reverting the changes to the original version, and edit the file with notepad++?
We should get only the changes you made, not the entire file.

Reply:
Done thank you